skademlia/client.go

```diff
@@ -183,12 +190,11 @@ func (c *Client) DialContext(ctx context.Context, addr string) (*grpc.ClientConn
 		return nil, errors.New("attempted to dial self")
 	}
 
-	c.peersLock.RLock()
+	c.peersLock.Lock()
 	if conn, exists := c.peers[addr]; exists {
-		c.peersLock.RUnlock()
+		c.peersLock.Unlock()
 		return conn, nil
 	}
-	c.peersLock.RUnlock()
```

Comment:
@iwasaki-kenta  thats cause of race. lock is unlocked after check but before map is populated with new connection, which allows to more than one goroutine to be created for same address